### PR TITLE
Deprecate `primaryColorBrightness`

### PR DESCRIPTION
**Pull request by guidezpl**
In line with https://flutter.dev/go/material-theme-system-updates and https://github.com/flutter/flutter/issues/91772, deprecates `primaryColorBrightness`. Usages in `AppBar` and `TextField` are removed.

A [Dart fix](https://dart.dev/tools/dart-fix) is provided. 

This is not a breaking change, per our [policy](https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes). However, should this causes any issues, the documentation for `primaryColorBrightness` explains alternatives for its former usages.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by guidezpl** [comment]
@HansMuller Can you please review before I write a breaking change doc?

File: packages/flutter/lib/src/material/text_field.dart [hunk 1]
```diff
@@ -1139,7 +1139,7 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
     final ThemeData theme = Theme.of(context);
     final TextSelectionThemeData selectionTheme = TextSelectionTheme.of(context);
     final TextStyle style = theme.textTheme.subtitle1!.merge(widget.style);
-    final Brightness keyboardAppearance = widget.keyboardAppearance ?? theme.primaryColorBrightness;
+    final Brightness keyboardAppearance = widget.keyboardAppearance ?? theme.colorScheme.brightness;
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/text_field.dart:1142]: You can use `theme.brightness here` (it means the same thing).

File: packages/flutter/lib/src/material/app_bar.dart [hunk 2]
```diff
@@ -1047,7 +1047,7 @@ class _AppBarState extends State<AppBar> {
       ? _systemOverlayStyleForBrightness(
           widget.brightness
           ?? appBarTheme.brightness
-          ?? theme.primaryColorBrightness,
+          ?? theme.primaryColorBrightness, // TODO: since backwardsCompatibility is deprecated, update to ThemeData.estimateBrightnessForColor(backgroundColor) or not?
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/app_bar.dart:1050]: Yes. We'll have to watch out for downstream problems.

File: packages/flutter/lib/src/material/theme_data.dart [hunk 3]
```diff
@@ -390,10 +394,10 @@ class ThemeData with Diagnosticable {
     final bool isDark = _brightness == Brightness.dark;
     primarySwatch ??= Colors.blue;
     primaryColor ??= isDark ? Colors.grey[900]! : primarySwatch;
-    primaryColorBrightness ??= estimateBrightnessForColor(primaryColor);
+    final Brightness _primaryBrightness = primaryColorBrightness ??= estimateBrightnessForColor(primaryColor);
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/theme_data.dart:397]: Prefer two assignment statements here.

**Review by HansMuller — APPROVED** [review]
LGTM

**Review by Piinks — APPROVED** [review]
LGTM! Thanks for writing dart fixes for this! :)

**Review by HansMuller — APPROVED** [review]
LGTM as well.

**Pull request merged**